src/core/qgsmaplayer.h

```diff
@@ -736,7 +737,8 @@ class CORE_EXPORT QgsMapLayer : public QObject
      * \param errorMsg this QString will be initialized on error
      * during the execution of writeSymbology
      */
-    virtual void exportNamedStyle( QDomDocument &doc, QString &errorMsg SIP_OUT ) const;
+    virtual void exportNamedStyle( QDomDocument &doc, QString &errorMsg SIP_OUT, QgsReadWriteContext &context,
+                                   QgsMapLayerStyle::StyleCategories categories = QgsMapLayerStyle::All ) const;
```

Comment:
Technically speaking, this is an API break (if someone subclassed and overwrote).
I guess a mapping could be done but it's definitely not trivial.

Question: is this even actively used? Then at least there would be an example on which to base this.

@nyalldawson @wonder-sk (others?)

Reply:
I don't think overriding layer methods is really intended use of the API. If we could, I'd like them to be marked "final" in the vector/raster classes. But sip doesn't like this, so we're forced with this annoying situation.

Anyway, I'm ok with breaking non-intended use of API.

Reply:
If it works, I'm more than happy to use it!